File: frontend/src/types/risk.ts
export type RiskSeverity = 'critical' | 'high' | 'medium' | 'low' | 'info'
export type RiskFindingStatus = 'open' | 'reviewed' | 'resolved' | 'dismissed'

export interface RiskFinding {
  id: string
  workspace_id: string
  document_id?: string | null
  primary_normalized_record_id?: string | null
  related_normalized_record_ids?: string[] | null
  risk_type: string
  severity: RiskSeverity
  title: string
  description?: string | null
  risk_data?: Record<string, unknown> | null
  status: RiskFindingStatus
  created_at: string
  updated_at: string
}

export interface RiskSummary {
  total: number
  critical: number
  high: number
  medium: number
  low: number
  info: number
  open: number
  reviewed: number
  resolved: number
  dismissed: number
}

export interface UpdateRiskFindingStatusRequest {
  status: RiskFindingStatus
}

export interface RunRiskCheckRequest {
  document_ids?: string[]
}

export interface RiskFindingListResponse {
  workspace_id: string
  total_records: number
  limit: number
  offset: number
  findings: RiskFinding[]
}

File: backend/app/schemas/risk.py
from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class RiskFindingResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    workspace_id: UUID
    document_id: UUID | None = None
    primary_normalized_record_id: UUID | None = None
    related_normalized_record_ids: list[str] | None = None
    risk_type: str
    severity: str
    title: str
    description: str
    risk_data: dict[str, Any] | None = None
    status: str
    created_at: datetime
    updated_at: datetime


class RiskFindingListResponse(BaseModel):
    workspace_id: UUID | None = None
    document_id: UUID | None = None
    total_records: int
    limit: int
    offset: int
    findings: list[RiskFindingResponse]


class RiskSummaryResponse(BaseModel):
    workspace_id: UUID
    total_findings: int
    counts_by_status: dict[str, int]
    counts_by_severity: dict[str, int]
    counts_by_risk_type: dict[str, int]


class RiskRunResponse(BaseModel):
    workspace_id: UUID
    total_findings: int
    summary: RiskSummaryResponse


class RiskFindingStatusUpdate(BaseModel):
    status: str
